indico/modules/events/editing/client/js/components/FileTypeModal.jsx

```diff
@@ -0,0 +1,104 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Form as FinalForm} from 'react-final-form';
+import {Button, Form, Header, Modal} from 'semantic-ui-react';
+
+import {FinalCheckbox, FinalField, FinalInput, FinalSubmitButton} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+import ExtensionList from './ExtensionList';
+
+export default function FileTypeModal({onClose, onSubmit, initialValues, header}) {
+  const handleSubmit = async (formData, form) => {
+    const error = await onSubmit(formData, form);
+    if (error) {
+      return error;
+    }
+    onClose();
+  };
+
+  return (
+    <FinalForm
+      onSubmit={handleSubmit}
+      subscription={{submitting: true}}
+      initialValues={initialValues}
+    >
+      {fprops => (
+        <Modal
+          onClose={onClose}
+          size="tiny"
+          closeIcon={!fprops.submitting}
+          closeOnDimmerClick={!fprops.submitting}
+          open
+        >
+          <Modal.Header content={header} />
+          <Modal.Content>
+            <Form id="file-type-form" onSubmit={fprops.handleSubmit}>
+              <FinalInput name="name" label={Translate.string('Name')} required />
+              <FinalField
+                name="extensions"
+                component={ExtensionList}
+                label={Translate.string('Extensions')}
+                isEqual={_.isEqual}
```

Comment:
You probably want `isEqual={unsortedArraysEqual}` here since the file type list isn't sorted so changing the order (e.g. by deleting in the middle and re-adding) doesn't need to mark the form as dirty.